pkg/updatechecker/updatechecker.go

```diff
@@ -183,18 +219,94 @@ type UpdateCheckRelease struct {
 // otherwise, if "IsAutomatic" is set to true (which means it's an automatic update check), then the version that matches the auto deploy configuration (if enabled) will be deployed.
 // returns the number of available updates.
 func CheckForUpdates(opts CheckForUpdatesOpts) (*UpdateCheckResponse, error) {
-	currentStatus, _, err := store.GetTaskStatus("update-download")
+	clientSet, err := k8sutil.GetClientset()
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get task status")
+		return nil, errors.Wrap(err, "failed to get clientset")
+	}
+
+	selectorLabels := map[string]string{
+		"metadata.name": "kotsadm-confg",
+	}
+	listOpts := metav1.ListOptions{
+		FieldSelector: labels.SelectorFromSet(selectorLabels).String(),
+	}
+
+	configMaps, err := clientSet.CoreV1().ConfigMaps(util.PodNamespace).List(context.TODO(), listOpts)
```

Comment:
Same, use `Get` instead of `List` and maybe move into store package

Reply:
looking into moving into store